booking_service.go

```diff
@@ -31,11 +38,28 @@ func main() {
 		return
 	}
 
-	// init db
-	err = dao.InitDB(configObj.Dsn)
+	var hostname string
+
+	var database Database
+	secret := os.Getenv("MYSQL_PASSWORD")
+	if secret != "" {
+		// Parse the JSON data into the struct
+		if err := json.Unmarshal([]byte(secret), &database); err != nil {
+			logrus.WithField("decodeSecretManager", database).Error("failed to decode value from secret manager")
+			return
+		}
+	}
+
+	if database.Password != "" {
+		hostname = database.Username + ":" + database.Password + "@tcp(ev-charger-mysql-db.cdklkqeyoz4a.ap-southeast-1.rds.amazonaws.com:3306)/evc?parseTime=true&charset=utf8mb4"
+	} else {
+		hostname = configObj.Dsn // localhost
+	}
+
+	err = dao.InitDB(hostname)
 	if err != nil {
 		logrus.WithField("config", configObj).Error("failed to connect to database")
-		return
+		//return
```

Comment:
if unable to connect we dun return? or panicc?

Reply:
The reason for this return is because of ZAP scanning. I am running localhost on GitHub pipeline, in order to scan dynamically.

If return or panic, local host won't be able to run without DB.

Reply:
yeah, i think is panic. But i cannot panic due to need to run localhost on pipeline for scanning.